auction-server/src/opportunity/service/handle_opportunity_bid.rs

```diff
@@ -0,0 +1,123 @@
+use {
+    super::{
+        make_adapter_calldata::MakeAdapterCalldataInput,
+        ChainTypeEvm,
+        Service,
+    },
+    crate::{
+        api::{
+            Auth,
+            RestError,
+        },
+        auction::{
+            handle_bid,
+            BidEvm,
+        },
+        opportunity::{
+            api::{
+                OpportunityBid,
+                OpportunityId,
+            },
+            contracts::{
+                erc20,
+                OpportunityAdapterErrors,
+            },
+        },
+    },
+    ethers::{
+        contract::ContractRevert,
+        types::Bytes,
+    },
+    time::OffsetDateTime,
+    uuid::Uuid,
+};
+
+pub struct HandleOpportunityBidInput {
+    pub opportunity_id:  OpportunityId,
+    pub opportunity_bid: OpportunityBid,
+    pub initiation_time: OffsetDateTime,
+    pub auth:            Auth,
+}
+
+fn parse_revert_error(revert: &Bytes) -> Option<String> {
+    let apdapter_decoded =
+        OpportunityAdapterErrors::decode_with_selector(revert).map(|decoded_error| {
+            format!(
+                "Opportunity Adapter Contract Revert Error: {:#?}",
+                decoded_error
+            )
+        });
+    let erc20_decoded = erc20::ERC20Errors::decode_with_selector(revert)
+        .map(|decoded_error| format!("ERC20 Contract Revert Error: {:#?}", decoded_error));
+    apdapter_decoded.or(erc20_decoded)
+}
+
+impl Service<ChainTypeEvm> {
+    pub async fn handle_opportunity_bid(
+        &self,
+        input: HandleOpportunityBidInput,
+    ) -> anyhow::Result<Uuid, RestError> {
```

Comment:
Let's use the same syntax everywhere:  `anyhow::Result<Uuid, RestError>` -> `Result<Uuid, RestError>`